scapy/utils.py

```diff
@@ -501,14 +501,21 @@ def do_graph(graph,prog=None,format=None,target=None,type=None,string=None,optio
             start_viewer = True
         else:
             with ContextManagerSubprocess("do_graph()"):
-                target = subprocess.Popen([conf.prog.display], stdin=subprocess.PIPE).stdin
+                target = subprocess.Popen([conf.prog.display],
+                                          stdin=subprocess.PIPE).stdin
     if format is not None:
         format = "-T%s" % format
     if isinstance(target, str):
-        target = open(os.path.abspath(target), "wb")
+        if target.startswith('|'):
+            target = subprocess.Popen(target[1:].lstrip(), shell=True,
+                                      stdin=subprocess.PIPE).stdin
+        elif target.startswith('>'):
+            target = open(target[1:].lstrip(), "wb")
+        else:
+            target = open(os.path.abspath(target), "wb")
     proc = subprocess.Popen("\"%s\" %s %s" % (prog, options or "", format or ""),
                             shell=True, stdin=subprocess.PIPE, stdout=target)
-    proc.communicate(input=raw(graph))
+    proc.stdin.write(raw(graph))
```

Comment:
You may have trouble with this test: because graphviz is not installed, the pipetool test with `graph` used to silently fail, but triggered coverage

Reply:
Will do, thanks!